packages/semantic-model/src/resolve.ts

```diff
@@ -298,3 +298,88 @@ const typesToIgnore: Record<string, undefined> = {
 function ignoreType(typeName: string): boolean {
   return has(typesToIgnore, typeName);
 }
+
+export function resolveTypeDefPropertyType(
```

Comment:
This looks very similar to `resolveType`. We should have only one function for resolving types unless there is a good reason to have multiple. As far as I know all `type` strings should have the same structure and should be processed the same way.